receiver/stanzareceiver/converter.go

```diff
@@ -0,0 +1,220 @@
+	logSlice := rls.InstrumentationLibraryLogs().At(0).Logs()
+
+	lr := pdata.NewLogRecord()
+	lr.InitEmpty()
+
+	lr.SetTimestamp(pdata.TimestampUnixNano(obsLog.Timestamp.UnixNano()))
+
+	sevText, sevNum := convertSeverity(obsLog.Severity)
+	lr.SetSeverityText(sevText)
+	lr.SetSeverityNumber(sevNum)
+
+	if len(obsLog.Labels) > 0 {
+		attributes := lr.Attributes()
+		for k, v := range obsLog.Labels {
+			attributes.InsertString(k, v)
+		}
+	}
+
+	lr.Body().InitEmpty()
+	insertToAttributeVal(obsLog.Record, lr.Body())
+
+	logSlice.Append(lr)
+
+	return out
+}
+
+func insertToAttributeVal(value interface{}, dest pdata.AttributeValue) {
+	switch t := value.(type) {
+	case bool:
+		dest.SetBoolVal(t)
+	case string:
+		dest.SetStringVal(t)
+	case []byte:
+		dest.SetStringVal(string(t))
+	case int64:
+		dest.SetIntVal(t)
+	case int32:
+		dest.SetIntVal(int64(t))
+	case int16:
+		dest.SetIntVal(int64(t))
+	case int8:
+		dest.SetIntVal(int64(t))
+	case int:
+		dest.SetIntVal(int64(t))
+	case uint64:
+		dest.SetIntVal(int64(t))
+	case uint32:
+		dest.SetIntVal(int64(t))
+	case uint16:
+		dest.SetIntVal(int64(t))
+	case uint8:
+		dest.SetIntVal(int64(t))
+	case uint:
+		dest.SetIntVal(int64(t))
+	case float64:
+		dest.SetDoubleVal(t)
+	case float32:
+		dest.SetDoubleVal(float64(t))
+	case map[string]interface{}:
+		dest.SetMapVal(toAttributeMap(t))
+	case []interface{}:
+		dest.SetMapVal(toAttributeMap(sliceToMap(t)))
+	default:
+		dest.SetStringVal(fmt.Sprintf("%v", t))
+	}
+}
+
+func toAttributeMap(obsMap map[string]interface{}) pdata.AttributeMap {
+	attMap := pdata.NewAttributeMap()
+	attMap.InitEmptyWithCapacity(len(obsMap))
+	for k, v := range obsMap {
+		switch t := v.(type) {
+		case bool:
+			attMap.InsertBool(k, t)
+		case string:
+			attMap.InsertString(k, t)
+		case []byte:
+			attMap.InsertString(k, string(t))
+		case int64:
+			attMap.InsertInt(k, t)
+		case int32:
+			attMap.InsertInt(k, int64(t))
+		case int16:
+			attMap.InsertInt(k, int64(t))
+		case int8:
+			attMap.InsertInt(k, int64(t))
+		case int:
+			attMap.InsertInt(k, int64(t))
+		case uint64:
+			attMap.InsertInt(k, int64(t))
+		case uint32:
+			attMap.InsertInt(k, int64(t))
+		case uint16:
+			attMap.InsertInt(k, int64(t))
+		case uint8:
+			attMap.InsertInt(k, int64(t))
+		case uint:
+			attMap.InsertInt(k, int64(t))
+		case float64:
+			attMap.InsertDouble(k, t)
+		case float32:
+			attMap.InsertDouble(k, float64(t))
+		case map[string]interface{}:
+			subMap := toAttributeMap(t)
+			subMapVal := pdata.NewAttributeValueMap()
+			subMapVal.SetMapVal(subMap)
+			attMap.Insert(k, subMapVal)
+		case []interface{}:
+			subMap := toAttributeMap(sliceToMap(t))
+			subMapVal := pdata.NewAttributeValueMap()
+			subMapVal.SetMapVal(subMap)
+			attMap.Insert(k, subMapVal)
+		default:
+			attMap.InsertString(k, fmt.Sprintf("%v", t))
+		}
+	}
+	return attMap
+}
+
+// This returns a map of stringified index to value,
+// rather than an array because the pdata package does not support arrays
```

Comment:
I think it does support arrays as of now.

Reply:
Thanks for pointing this out. I missed the update. 

I've switched to using the new array functionality now.